src/super_gradients/training/models/segmentation_models/unet/unet_decoder.py

```diff
@@ -34,8 +35,16 @@ class UpFactorBlock(AbstractUpFuseBlock):
     Ignore Skip features, simply apply upsampling and ConvBNRelu layers.
     """
 
-    def __init__(self, in_channels: int, skip_channels: int, out_channels: int, up_factor: int, mode: str, num_repeats: int, **kwargs):
+    def __init__(self, in_channels: int, skip_channels: int, out_channels: int, up_factor: int, mode: str, num_repeats: int, fallback_mode: str, **kwargs):
```

Comment:
Two questions here:
1) Why not make/annotate `mode` & `fallback_mode` of `UpsampleMode` type?
2) Why do we prefer `fallback_mode` over raising an exception if requested mode is not compatible?

Reply:
1. I agree we can hint a type `Union[str, UpsampleMode]`.  Passing `str` should also be an option IMO, it simplify the yaml init, and it's common usage as torch reduction modes and upsample modes i.e
2. Let's take an example of a module with 4 decoder stages, in case only one of the stages isn't supported for pixel-shuffle, this module is still a valid network and the user might want to use this combination.
We can make this argument `Optional[str] = None`, in case is None, we can raise an axception

Reply:
Changed hit type to `Union[UpsampleMode, str]` [bb14e68](https://github.com/Deci-AI/super-gradients/pull/707/commits/bb14e6877d8e24fb2d7de4d74b91d75b892b496f)